include/cuco/detail/open_addressing/kernels.cuh

```diff
@@ -356,6 +356,96 @@ CUCO_KERNEL __launch_bounds__(BlockSize) void find(InputIt first,
   }
 }
 
+/**
+ * @brief Inserts all elements in the range `[first, last)`.
+ *
+ * @note: For a given element `*(first + i)`, if the container doesn't already contain an element
+ * with an equivalent key, inserts the element at a location pointed by `iter` and writes
+ * `iter` to `found_begin + i` and writes `true` to `inserted_begin + i`. Otherwise, finds the
+ * location of the equivalent element, `iter` and writes `iter` to `found_begin + i` and writes
+ * `false` to `inserted_begin + i`.
+ *
+ * @tparam CGSize Number of threads in each CG
+ * @tparam BlockSize Number of threads in each block
+ * @tparam InputIt Device accessible input iterator whose `value_type` is
+ * convertible to the `value_type` of the data structure
+ * @tparam FoundIt Device accessible random access output iterator whose `value_type`
+ * is constructible from `map::iterator` type
+ * @tparam Boolt Device accessible random access output iterator whose `value_type`
+ * is constructible from `bool`
+ * @tparam Ref Type of non-owning device container ref allowing access to storage
+ *
+ * @param first Beginning of the sequence of input elements
+ * @param n Number of input elements
+ * @param found_begin Beginning of the sequence of elements found for each key
+ * @param inserted_begin Beginning of the sequence of booleans for the presence of each key
+ * @param ref Non-owning container device ref used to access the slot storage
+ */
+template <int32_t CGSize,
+          int32_t BlockSize,
+          typename InputIt,
+          typename FoundIt,
+          typename Boolt,
```

Comment:
nit: according to our naming conventions this should be `BoolT`. However, I would prefer `InsertedIt`. What do you think?

Reply:
Yes, `InsertedIt` sounds better!